src/marked.js

```diff
@@ -139,45 +139,99 @@ marked.defaults = defaults;
  */
 
 marked.use = function(extension) {
-  const opts = merge({}, extension);
-  if (extension.renderer) {
-    const renderer = marked.defaults.renderer || new Renderer();
-    for (const prop in extension.renderer) {
-      const prevRenderer = renderer[prop];
-      renderer[prop] = (...args) => {
-        let ret = extension.renderer[prop].apply(renderer, args);
-        if (ret === false) {
-          ret = prevRenderer.apply(renderer, args);
+  if (!Array.isArray(extension)) { // Wrap in array if not already to unify processing
+    extension = [extension];
+  }
+
+  const opts = merge({}, ...extension);
+  const extensions = marked.defaults.extensions || { renderers: {}, walkableTokens: {} };
+  let hasExtensions;
+
+  extension.forEach((pack) => {
+    // ==-- Parse "addon" extensions --== //
+    if (pack.extensions) {
+      hasExtensions = true;
+      pack.extensions.forEach((ext) => {
+        if (ext.renderer && ext.name) { // Renderers must have 'name' property
+          extensions.renderers[ext.name] = ext.renderer;
```

Comment:
Do we want to fallback to extension renderer with same name if exists on `false`. I don't know if we need to but thought I'd put the idea out there. Same with extension tokenizer.

Reply:
Maybe. In case multiple packages happen to choose the same extension name?

Reply:
Ya ! could see having an extension that provides a special case for another extension.

Reply:
I set the renderers to fall back if there is another extension with the same name.

The tokenizers however are just piled into an array `extensions.block` or `extensions.inline` and executed in array order. This means if a tokenizer returns false, it will just try the next tokenizer in the array anyway. I did change the append order from `extensions.block.push(newTokenizer)` to `extensions.block.unshift(newTokenizer)` so extensions later in the load order are given first priority.

So instead of falling back directly, tokenizers will just rely on load order, which seems logical to me. Assume `extension1_false` has the same `name` as `extension1` but always returns false:
```
marked.use([extension1, extension2, extension1_false]); \\Latest load gets priority

// In Lexer.js:
if (token = extension1_false.call(this, src, tokens)) {... 
\\ returns false

if (token = extension2.call(this, src, tokens)) {... 
\\ doesn't match

if (token = extension1.call(this, src, tokens)) {...
\\ Matches
```